src/server/nonClassEvent/nonClassEvent.controller.ts

```diff
@@ -49,4 +60,23 @@ export class NonClassEventController {
       );
     return nonClassEvents;
   }
+
+  @Post('/')
```

Comment:
This is missing docs, both in docblock form and in `@Api` tags from swagger describing the body, return values, exceptions, etc.

It should probably have a `RequireGroup` guard restriction, though I don't think we have a group that matches the use case at present -- I _think_ we had talked about letting no-Admin users create their own NonClassEvents that would need to be approved. Maybe we can temporarily add a `GROUP.NON_CLASS` until we have more clarity on that.

Reply:
> This is missing docs, both in docblock form and in `@Api` tags from swagger describing the body, return values, exceptions, etc.

Well spotted. I've fixed that in ca24bc118e91121ccbdb4f9d13e2ea4bcb427522 as well as updating the `CreateNonClassParentDTO` in ae561bd7523f425e8ac7f3ea6f07bfa8430694e8 to denote the optional properties as such.

> Maybe we can temporarily add a `GROUP.NON_CLASS` until we have more clarity on that.

Added that in 244bcb5559b847d1d3e24f86dcf1cc4b73331221 (despite the commit message being a little off - I originally set it to `GROUP.ADMIN` and then amended it before pushing)

Reply:
Also updated my tests in 9e3a0707f8a02ca5a5ad3f98a79f186bf7aec130 to force `authStub` to return a user with the correct group membership to edit non-class events(otherwise, the tests fail).

This also involved the creation of said user in our test data, so we have that going forward.